src/ui/public/state_management/state.js

```diff
@@ -21,20 +22,23 @@ export default function StateProvider(Private, $rootScope, $location, config) {
   function State(urlParam, defaults, { hashingStore, notifier } = {}) {
     State.Super.call(this);
 
-    let self = this;
-    self.setDefaults(defaults);
-    self._urlParam = urlParam || '_s';
+    this.setDefaults(defaults);
+    this._urlParam = urlParam || '_s';
     this._notifier = notifier || new Notifier();
-    self._hashingStore = hashingStore || new HashingStore({
-      store: new LazyLruStore({
+
+    this._hashingStore = hashingStore || (() => {
+      const lazyLruStore =  new LazyLruStore({
         id: `${this._urlParam}:state`,
         store: window.sessionStorage,
         maxItems: MAX_BROWSER_HISTORY
-      })
-    });
+      });
+
+      return new HashingStore(createStorageHash, lazyLruStore);
```

Comment:
We can simplify the constructor to the point where it doesn't require a config object.